### PR TITLE
Issue #13043: Modify link javadoc tag reference to be optional

### PR DESCRIPTION
**Pull request by 0xbakry**
Solves https://github.com/checkstyle/checkstyle/issues/13043

CLI:

```
$ cat config1.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name = "Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
       <module name="NonEmptyAtclauseDescription"/>
    </module>

</module>

$ cat Test1.java
public class Test1 {

  /**
   * {@link}
   * {@linkplain}
   */
  public void method(String a)
  {

  }
}
```

```
$ java -jar checkstyle-10.10.1-SNAPSHOT-all.jar -c config1.xml Test1.java
Starting audit...
Audit done.

```

Diff Regression projects: https://gist.githubusercontent.com/0xbakry/7853843aa3d69aee301d7a8d62d08ebb/raw/65a7d6bee0ad8ad789c17f778ce7d8b0056ae28b/my_projects
Diff Regression config: https://gist.githubusercontent.com/0xbakry/a1fcb13eef42f29de8ff500892f34166/raw/beceb87d12215501a53ffc29e87f6c0af78f588c/config_check.xml

### CODE REVIEW

**Comment by 0xbakry** [comment]
GitHub, generate report

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocJavadocTagsWithoutArgs.java [hunk 1]
```diff
@@ -59,14 +59,14 @@ private void readObject(java.io.ObjectInputStream inputStream) {
     }
 
     /**
-     * {@link} // violation
+     * {@link} // ok
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocJavadocTagsWithoutArgs.java:62]: Unfortunately this doesn't show practical reason to user. As it is not real Check, change from violation to ok , does not mean much.

https://checkstyle.org/config_javadoc.html#SingleLineJavadoc
https://checkstyle.org/config_javadoc.html#SummaryJavadoc

Please extend Inputs for such Checks to show what type of possibilities your fix is opening.

**Review by romani — CHANGES_REQUESTED** [review]
Items:

**Comment by nrmancuso** [comment]
@0xbakry do you plan to continue work on this PR?

**Comment by 0xbakry** [comment]
> @0xbakry do you plan to continue work on this PR?

Yeah, I'm a little busy with some work right now, but I'm planning to continue working on the project soon.

**Comment by rnveach** [comment]
Needs to be rebased.

**Comment by romani** [comment]
if no activity before  1 october , we will close PR to let somebody else to finish this update by reusage of this code 

@0xbakry , please find time to finish update, looks like a little is left to be done

**Comment by 0xbakry** [comment]
Sorry for the delay, I don't have access to my laptop most of the time, so I'm closing this PR to let someone else work on it.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocJavadocTagsWithoutArgs.java (hunk 1 above)
Comment (0xbakry) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocJavadocTagsWithoutArgs.java]: Hey, I know this was a big delay from my side.
but sorry I didn't get how this doesn't mean much? there were several parsing errors, and as I see they were solved.
**I know I need to push a single commit and I'm doing this now** I'm just talking about the updated grammar

**Comment by romani** [comment]
Here is response https://github.com/checkstyle/checkstyle/pull/13046#discussion_r1564931186 from author, let's wait a bit more. Thanks a lot for coming back to it.

**Comment by romani** [comment]
@0xbakry , please find time to finish PR

**Pull request closed without merging**